components/movementsensor/utils.go

```diff
@@ -90,6 +90,12 @@ type LastPosition struct {
 	mu           sync.Mutex
 }
 
+// LastCompassHeading store the last valid compass heading seen by the movement sensor.
+type LastCompassHeading struct {
+	lastcompassheading float64
+	mu                 sync.Mutex
+}
```

Comment:
Discussion question: you've grouped all the `type...struct` lines together, and then put all the functions on all the types below. Personally, I'd prefer to group all the `LastCompassHeading` code together, and all the `LastPosition` code together, etc., so I don't need to search for extra functionality on this struct elsewhere. What do others prefer? My hope is that the majority likes one way or the other, and then we'll know to write our code in that style in the future. (and it's possible I'm not in the majority and I should switch the code I write instead.)